provider/azure/internal/azureauth/interactive.go

```diff
@@ -0,0 +1,347 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package azureauth
+
+import (
+	"fmt"
+	"io"
+	"net/url"
+	"path"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/arm/authorization"
+	"github.com/Azure/azure-sdk-for-go/arm/resources/subscriptions"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/juju/errors"
+	"github.com/juju/loggo"
+	"github.com/juju/utils"
+	"github.com/juju/utils/clock"
+
+	"github.com/juju/juju/provider/azure/internal/ad"
+	"github.com/juju/juju/provider/azure/internal/errorutils"
+	"github.com/juju/juju/provider/azure/internal/tracing"
+)
+
+var logger = loggo.GetLogger("juju.provider.azure.internal.azureauth")
+
+const (
+	// jujuApplicationId is the ID of the Azure application that we use
+	// interactive authentication. When the user logs in, a service
```

Comment:
"that we use interactive authenthication." reads funny... Is there a chance to rephrase?